scripts/dts/edtlib.py

```diff
@@ -693,26 +687,55 @@ def _init_clocks(self):
             else:
                 clock.frequency = None
 
-            self.clocks.append(clock)
-
-        _add_names(node, "clock-names", self.clocks)
-
     def _init_pwms(self):
         # Initializes self.pwms
 
-        self.pwms = []
+        self.pwms = self._simple_phandle_val_list("pwm", PWM)
 
-        for controller_node, spec in _pwms(self._node):
-            controller = self.edt._node2dev[controller_node]
+    def _simple_phandle_val_list(self, name, cls):
+        # Helper for parsing properties like
+        #
+        #     <name>s = <phandle value phandle value ...>
+        #     (e.g., gpios = <&foo 1 2 &bar 3 4>)
+        #
+        # , where each phandle points to a node that has a
+        #
+        #     #<name>-cells = <size>
+        #
+        # property that gives the number of cells in the value after the
+        # phandle. Also parses any
+        #
+        #     <name>-names = "...", "...", ...
+        #
+        # Arguments:
+        #
+        # name:
+        #   The <name>, e.g. "clock" or "pwm". Note: no -s in the argument.
+        #
+        # cls:
+        #   A class object. Instances of this class will be created and the
+        #   'dev', 'controller', 'specifier', and 'name' fields initialized.
+        #   See the documentation for e.g. the PWM class.
+        #
+        # Returns a list of 'cls' instances.
 
-            pwm = PWM()
-            pwm.dev = self
-            pwm.controller = controller
-            pwm.specifier = self._named_cells(controller, spec, "pwms")
+        prop = self._node.props.get(name + "s")
```

Comment:
I don't think we should add the 's' here.  I think its a bit confusing on the caller and we should leave it to the caller to pass the name as it is in the .dts

Reply:
Had it like that at first, but it gets ugly in that you'd instead have to strip off a last character that may or may not be an `s` (and maybe add an ugly `assert` for it), so this way seemed nicer and cleaner. The ends-with-an-`s` requirement is enforced more directly.

Note that it's documented in the docstring.